src/test/test-dns-domain.c

```diff
@@ -734,8 +734,10 @@ TEST(dns_name_is_valid_or_address) {
         assert_se(dns_name_is_valid_or_address("foobar..com") == 0);
         assert_se(dns_name_is_valid_or_address("foobar.com.") > 0);
         assert_se(dns_name_is_valid_or_address("127.0.0.1") > 0);
+        assert_se(dns_name_is_valid_or_address("127.0.0.1:8080") > 0);
         assert_se(dns_name_is_valid_or_address("::") > 0);
         assert_se(dns_name_is_valid_or_address("::1") > 0);
+        assert_se(dns_name_is_valid_or_address("[::1]:8080") > 0);
```

Comment:
uh, how could that possibly succeed? this is weird. the function is supposed to valid DNSnames or IP addresses, but not a port number

Reply:
Thank you for your feedback I hope it makes this MR better and into a state you can accept.

The underlying call to glibc's inet_pton() thinks this is valid. Maybe `manager_parse_server_string` in `src/timesync/timesyncd-conf.c` should reject a string containing one colon (for server.domain:port or I.P.v.6:port) or substring ']:' (for [IP::v:6]:port), but instead you got this MR and attempt to parse and handle non-standard port numbers.

This configuration flowing through to [manager_connect:L880](https://github.com/systemd/systemd/blob/main/src/timesync/timesyncd-manager.c#L880) was why I raised #32218.